platform-modules/batch-models/src/main/scala/org/ekstep/analytics/model/DeviceSummaryModel.scala

```diff
@@ -65,22 +78,32 @@ object DeviceSummaryModel extends IBatchModelTemplate[String, DeviceInput, Devic
             val dial_success = dialcodes_events.filter(f => f.edata.size > 0).length
             val dial_failure = dialcodes_events.filter(f => f.edata.size == 0).length
             val content_downloads = raw.filter(f => "INTERACT".equals(f.eid)).length
-            (index, DeviceSummary(index.device_id, index.channel, CommonUtil.roundDouble(total_ts, 2), total_launches, contents_played, unique_contents_played, content_downloads, DialStats(dial_count, dial_success, dial_failure), DtRange(startTimestamp, endTimestamp), syncts, startTimestamp))
+            (index, DeviceSummary(index.device_id, index.channel, CommonUtil.roundDouble(total_ts, 2), total_launches, contents_played, unique_contents_played, content_downloads, DialStats(dial_count, dial_success, dial_failure), DtRange(startTimestamp, endTimestamp), syncts, firstAccess))
         }
-        val firstAccessFromCassandra = summary.map{ x => x._1}
-          .joinWithCassandraTable[Option[Long]](Constants.DEVICE_KEY_SPACE_NAME, Constants.DEVICE_PROFILE_TABLE).select("first_access")
-          .on(SomeColumns("device_id"))
-        summary.leftOuterJoin(firstAccessFromCassandra)
-          .map{ x =>
-              val firstAccessValue = x._2._2.getOrElse(Option(0L))
-              if(firstAccessValue.getOrElse(0L) != 0)
-                  x._2._1.copy(firstAccess = firstAccessValue.getOrElse(x._2._1.firstAccess))
-              else x._2._1
+
+        implicit val sqlContext = new SQLContext(sc)
+        val responseDf = sqlContext.sparkSession.read.jdbc(url, Constants.DEVICE_PROFILE_TABLE, connProperties).select("device_id","first_access")
+        responseDf.show()
+        val encoder = Encoders.product[FirstAccessByDeviceID]
+        val firstAccessRDD = responseDf.as[FirstAccessByDeviceID](encoder).rdd
+
+        val postgresData = firstAccessRDD.map(f => {(f.device_id, f.first_access)})
+        val summaryData = summary.map(f => (f._1.device_id, f._2))
+        summaryData.leftOuterJoin(postgresData)
+          .map{f =>
+              val defaultDate = CommonUtil.getTimestampFromEpoch(0L)
+          val firstAccessVal = f._2._2.getOrElse(Option(defaultDate))
+          if(firstAccessVal != defaultDate) {
+              f._2._1.copy(firstAccess = firstAccessVal.getOrElse(f._2._1.firstAccess))
+          }
+          else
+              f._2._1
```

Comment:
This logic doesn't seem right. Here are the scenarios:
1. if firstAccess in db is null - update with the value from the summary event
2. if firstAccess in db is not null - compare this with the value from summary event and use the lowest date.

I don't seem to see the second logic in place

Reply:
It was not in the previous implementation. I have updated the changes